modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/di/ExperimentModule.kt

```diff
@@ -0,0 +1,64 @@
+package au.com.shiftyjelly.pocketcasts.analytics.di
+
+import android.content.Context
+import au.com.shiftyjelly.pocketcasts.analytics.AccountStatusInfo
+import au.com.shiftyjelly.pocketcasts.analytics.experiments.Experiment
+import au.com.shiftyjelly.pocketcasts.analytics.experiments.ExperimentLogger
+import au.com.shiftyjelly.pocketcasts.analytics.experiments.ExperimentProvider
+import au.com.shiftyjelly.pocketcasts.servers.di.Cached
+import com.automattic.android.experimentation.VariationsRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import java.io.File
+import javax.inject.Singleton
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import okhttp3.OkHttpClient
+
+@Module
+@InstallIn(SingletonComponent::class)
+object ExperimentModule {
+
+    @Provides
+    @Singleton
+    fun provideExperimentProvider(
+        accountStatusInfo: AccountStatusInfo,
+        repository: VariationsRepository,
+    ): ExperimentProvider = ExperimentProvider(accountStatusInfo, repository)
+
+    @Provides
+    @Singleton
+    internal fun provideExperimentLogger(): ExperimentLogger {
+        return ExperimentLogger()
+    }
+
+    @Provides
+    @Singleton
+    fun provideVariationsRepository(
+        @ApplicationContext context: Context,
+        logger: ExperimentLogger,
+        @Cached okHttpClient: OkHttpClient,
+    ): VariationsRepository {
+        val cacheDir = File(context.cacheDir, "experiments_cache").apply {
+            if (!exists()) mkdirs()
+        }
```

Comment:
@wzieba Do you know if `VariationsRepository` repository handles file creation? I'm asking because we shouldn't be doing IO operations on the main thread nor in the DI graph.

Reply:
At this moment, no, it doesn't. I can add such an addition, though.

But I wonder, wouldn't simple `cacheDir` work fine here?

```suggestion
        val cacheDir = ontext.cacheDir
```

WDYT?

Reply:
Using `cacheDir` would be fine but this directory can also not exists.

Reply:
Also, a separate point. But after thinking about it I don't think we should be using `cacheDir` for anything related to experiments. This directory can be easily wiped by the system. We should use something less volatile like `filesDir`.

Reply:
Great point about `filesDir`. That makes sense to me. What do you think @wzieba ?

Reply:
Sounds good to me 👍 The ExPlat Android SDK should handle a case with removed cache as well, but it's still good to make maximum use of it (use cache as long as it's valid). Especially, that the cache is so tiny.

Related to this, I'm adding an improvement: https://github.com/Automattic/Automattic-Tracks-Android/pull/248, so you could easily provide cache directory as a subdirectory of `filesDir`. The improvement should be available soon!

Reply:
Tracks `6.0.2` is now available that covers a case of non-existing parent directory 👍